File: TheaterSuggestions/CSharp/SeatsSuggestions.Domain/PartyRequested.cs
using System;
using System.Collections.Generic;
using Value;

namespace SeatsSuggestions.Domain
{
    public class PartyRequested : ValueType<PartyRequested>
    {
        public PartyRequested(int partySize)
        {
            if (partySize <= 0 || partySize > 20)
                throw new ArgumentException($"{nameof(partySize)} should be greater than zero and less than 20");

            PartySize = partySize;
        }

        public int PartySize { get; }

        protected override IEnumerable<object> GetAllAttributesToBeUsedForEquality()
        {
            return new object[] { PartySize };
        }
    }
}

File: TheaterSuggestions/Java/SeatsSuggestionsDomain/src/main/java/com/baasie/SeatsSuggestionsDomain/PartyRequested.java
package com.baasie.SeatsSuggestionsDomain;

import lombok.EqualsAndHashCode;

@EqualsAndHashCode
public class PartyRequested {

    public static final int MAX_PARTY_SIZE = 6;
    private final int _partySize;

    public PartyRequested(int partySize)
    {
        if (partySize <= 0 || partySize > MAX_PARTY_SIZE)
            throw new IllegalArgumentException("Parameter 'partySize' should be greater than zero and less than 6");

        _partySize = partySize;
    }

    public int partySize() { return _partySize; }
}
